integration-libs/punchout/root/services/punchout-auth-http-header.service.ts

```diff
@@ -50,37 +47,21 @@ export class PunchoutAuthHttpHeaderService extends AuthHttpHeaderService {
   /**
    * @override
    *
-   * On backend errors indicating expired `refresh_token`, punchout session gets ended, user is redirected to login page
+   * On backend errors indicating expired `refresh_token`, 2 punchout use cases:
+   * - When initializing punchout session, previous token gets silently revoked, punchoutFacade can then create punchout session.
+   * - When punchout session already exists, punchout session gets ended.
    * It is a workaround to address CXSPA-9608 - Public pages not displayed when token is invalid.
    * To be removed once CXSPA-9608 is closed.
    */
   public handleExpiredRefreshToken(): void {
-    if (
-      this.punchoutDetectionService.isPunchoutSession() ||
-      this.punchoutDetectionService.isPunchoutSessionPage()
-    ) {
-      this.punchoutFacade.endPunchoutSession().subscribe();
-    } else {
-      super.handleExpiredRefreshToken();
+    if (this.punchoutDetectionService.isPunchoutSessionPage()) {
+      this.punchoutFacade.logoutPunchoutUser().subscribe();
+      return;
     }
-  }
-
-  /**
-   * @override
-   * Refreshes access_token and then retries the call with the new token.
-   * When punchout session exists,  punchout session gets ended, user is redirected to login page
-   */
-  public handleExpiredAccessToken(
```

Comment:
Redundancy. UI action need to be done on handleExpiredRefreshToken (not handleExpiredAccessToken which is an interceptor)
It is not needed anymore since we refactored punchoutService.getPunchoutSession where we now always logout user before fetching occ api punchout session